src/App.js

```diff
@@ -1,15 +1,350 @@
-import React, { Component } from 'react';
+import React, {Component} from 'react';
+import uuid from 'uuid';
+import update from 'immutability-helper';
+import moment from 'moment';
+import axios from 'axios';
+
+import ContactList from './ContactList';
+import SearchBar from './SearchBar';
+import ActionHistory from './ActionHistory';
+import ToggleableContactForm from './ToggleableContactForm';
+
 
 class App extends Component {
+
+  constructor() {
+
+    super();
+
+    this.state = {
+      searchText: '',
+      contact: this.buildNewContact(),
+      validationErrors: {},
+      contacts: [],
+      selectedContacts: [],
+      backupContacts: [],
+      actionHistory: [],
+    };
+
+    // Done in constructor instead of in onEvent(s) for performance,
+    // if passed in the bind will be redone every render?
+    this.handleSearchBarChange = this.handleSearchBarChange.bind(this);
+    this.handleAddContactSubmit = this.handleAddContactSubmit.bind(this);
+    this.onInputChange = this.onInputChange.bind(this);
+    this.handleResetClick = this.handleResetClick.bind(this);
+    this.handleRemoveActionFromHistoryClick = this.handleRemoveActionFromHistoryClick.bind(this);
+    this.handleRemoveSelectedClick = this.handleRemoveSelectedClick.bind(this);
+    this.handleAddToSelectedClick = this.handleAddToSelectedClick.bind(this);
+    this.removeStaleActionHistory = this.removeStaleActionHistory.bind(this);
```

Comment:
Nice job on avoiding the creation of a new function on each render call. 
You can actually avoid all those calls in the constructor when you define them on the class with ES6 syntax: 
``` 
class Something {
  myBoundFunction = (param1) => {
     // Here `this` is always the component instance
  }
}
```